src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,140 @@
+
+export interface PaginationProps {
+  /**
+   * Unique id for component testing
+   */
+  id: string;
+  /**
+   * Base class for Pagination component.
+   */
+  className?: string;
+  /**
+   * Child element pass in to display as options to select from.
+   */
+  children?: React.ReactNode;
+  /**
+   * Boolean to specify whether the `<Pagination>` should be disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Optional text to display when disable is true
+   */
+  disabledText?: string;
+  /**
+   * Optional `onClick` handler that is called whenever the `<Pagination>` value is clicked
+   */
+  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  /**
+   * Optional `onChange` handler that is called whenever the `<Pagination>` value is changed
+   */
+  onChange?: (selectedValue: string) => void;
+  /**
+   * Specify the value of the `<Pagination>`
+   */
+  value?: string | undefined;
+  /**
+   * The default Inline style to apply to the `<Pagination>` Select component
+   */
+  variant?: 'inline';
+}
+
+/**
+ * ## Overview
+ *
+ * A component for adding a Pagination component.
+ *
+ * [Figma Link](https://www.figma.com/design/hxqgsE26wM7hII0WaUaDfF/RW---TIMED-Lot-Details-(PDP)?node-id=1-2&node-type=CANVAS&m=dev)
+ *
+ * [Storybook Link](https://phillips-seldon.netlify.app/?path=/docs/components-pagination--overview)
+ */
+
+const Pagination = ({
+  className,
+  onChange,
+  onClick,
+  variant = 'inline',
+  disabledText = 'Lot',
+  children,
+  ...props
+}: PaginationProps) => {
+  const type = 'pagination';
+  const { className: baseClassName, ...commonProps } = getCommonProps(props, 'Pagination');
+  const { id, value, isDisabled } = props;
+  const [selectedValue, setSelectedValue] = React.useState(value);
+  const selectRef = React.useRef<HTMLSelectElement>(null);
+
+  const paginationClick = (e: React.MouseEvent<HTMLElement>) => {
+    const buttonType = e.currentTarget.dataset.testid;
+    const selectedIndex = selectRef.current?.selectedIndex;
+    const firstIndexValue = selectRef.current?.options[0]?.value;
+    const lastIndex = selectRef.current?.childElementCount ? selectRef.current?.childElementCount - 1 : 0;
+    const lastIndexValue = selectRef.current?.options[lastIndex]?.value;
+    if (selectedIndex !== undefined && firstIndexValue && lastIndexValue) {
+      const prevValue = selectRef.current?.options[selectedIndex - 1]?.value;
+      const nextValue = selectRef.current?.options[selectedIndex + 1]?.value;
+      if (buttonType === `${id}-previous-button`)
+        prevValue ? setSelectedValue(prevValue) : setSelectedValue(lastIndexValue);
+      else if (buttonType === `${id}-next-button`)
+        nextValue ? setSelectedValue(nextValue) : setSelectedValue(firstIndexValue);
+    }
+  };
+  const paginationSelect = (e: Event) => {
+    const target = e.target as HTMLSelectElement;
+    setSelectedValue(target.value);
+  };
+
+  React.useEffect(() => {
+    selectedValue && onChange && onChange(selectedValue);
```

Comment:
can we remove this `useEffect` and just trigger the `onChange` callback separately in both `paginationSelect` and `paginationClick`  I try to reduce this as much as possible.  (For instance on first render, it's going to trigger this `onChange` even though nothing has changed with the `useEffect` like this.